### PR TITLE
service: am: Add support for LLE Mii Edit Applet

### PR DESCRIPTION
**Pull request by german77**
After more than 6 months of research, 15 PR, a few thousand lines of code and the holy illumination from byte. I'm happy to announce that yuzu can now boot LLE applets. This PR allows to boot the mii editor from firmware via `tools->mii editor`. 

You can create, edit or even load your mii from an amiibo using your joycon. Using them in games or anything related. All features are fully supported with exception of send/receive from other consoles.

Closes: #11280

image
Finally thank you to all yuzu team for making this PR possible. (Forgive me if I forgot someone)

### CODE REVIEW

File: src/core/hle/service/am/am.cpp [hunk 1]
```diff
@@ -2064,4 +2217,46 @@ IProcessWindingController::IProcessWindingController(Core::System& system_)
 }
 
 IProcessWindingController::~IProcessWindingController() = default;
+
+void IProcessWindingController::GetLaunchReason(HLERequestContext& ctx) {
+    LOG_WARNING(Service_AM, "called");
```
Comment (FearlessTobi) [inline src/core/hle/service/am/am.cpp:2222]: stubbed?

File: src/core/hle/service/am/am.cpp [hunk 2]
```diff
@@ -2064,4 +2217,46 @@ IProcessWindingController::IProcessWindingController(Core::System& system_)
...
+    IPC::ResponseBuilder rb{ctx, 3};
+    rb.Push(ResultSuccess);
+    rb.PushRaw(reason);
+}
+
+void IProcessWindingController::OpenCallingLibraryApplet(HLERequestContext& ctx) {
+    const auto applet_id = Applets::AppletId::MiiEdit;
+    const auto applet_mode = Applets::LibraryAppletMode::AllForeground;
+
+    LOG_WARNING(Service_AM, "called with applet_id={:08X}, applet_mode={:08X}", applet_id,
```
Comment (FearlessTobi) [inline src/core/hle/service/am/am.cpp:2244]: either log_info or a stub warning if this isn't fully implemented

File: src/core/hle/service/ns/iplatform_service_manager.cpp [hunk 3]
```diff
@@ -262,8 +262,17 @@ void IPlatformServiceManager::GetSharedMemoryNativeHandle(HLERequestContext& ctx
 }
 
 void IPlatformServiceManager::GetSharedFontInOrderOfPriority(HLERequestContext& ctx) {
+    // The maximum number of elements that can be returned is 6. Regardless of the available fonts
+    // or buffer size-
```
Comment (FearlessTobi) [inline src/core/hle/service/ns/iplatform_service_manager.cpp:266]:
```suggestion
    // or buffer size.
```

File: src/yuzu/main.cpp [hunk 4]
```diff
@@ -4134,6 +4135,27 @@ void GMainWindow::OnToggleStatusBar() {
     statusBar()->setVisible(ui->action_Show_Status_Bar->isChecked());
 }
 
+void GMainWindow::OnMiiEdit() {
+    auto bis_system = system->GetFileSystemController().GetSystemNANDContents();
```
Comment (FearlessTobi) [inline src/yuzu/main.cpp:4139]: can we grey out the ui option if we detect this is not available?

File: src/yuzu/main.cpp [hunk 5]
```diff
@@ -4134,6 +4135,27 @@ void GMainWindow::OnToggleStatusBar() {
...
+
+    auto mii_applet_nca =
+        bis_system->GetEntry(0x0100000000001009ull, FileSys::ContentRecordType::Program);
+    if (!mii_applet_nca) {
+        QMessageBox::warning(this, tr("Mii Edit Applet"),
+                             tr("Mii editor is not available. Please reinstall firmware."));
+        return;
+    }
+
+    QString filename = QString::fromStdString((mii_applet_nca->GetFullPath()));
```
Comment (FearlessTobi) [inline src/yuzu/main.cpp:4154]:
```suggestion
    const auto filename = QString::fromStdString((mii_applet_nca->GetFullPath()));
```

File: src/yuzu/main.ui [hunk 6]
```diff
@@ -368,6 +370,11 @@
     <string>&amp;Capture Screenshot</string>
    </property>
   </action>
+  <action name="action_Load_Mii_Edit">
+    <property name="text">
+      <string>&amp;Mii Editor</string>
```
Comment (FearlessTobi) [inline src/yuzu/main.ui:375]:
```suggestion
      <string>&amp;Open Mii Editor</string>
```

**Review by FearlessTobi — APPROVED** [review]
No tobi mii...

51pkHtmKkvL _AC_UF894,1000_QL80_

File: src/core/hle/service/am/am.cpp [hunk 7]
```diff
@@ -1361,10 +1382,142 @@ ILibraryAppletSelfAccessor::ILibraryAppletSelfAccessor(Core::System& system_)
...
+    };
+    // clang-format on
+
+    RegisterHandlers(functions);
+}
+
+IAppletCommonFunctions::~IAppletCommonFunctions() = default;
+
+void IAppletCommonFunctions::SetCpuBoostRequestPriority(HLERequestContext& ctx) {
+    LOG_ERROR(Service_AM, "called");
```
Comment (FearlessTobi) [inline src/core/hle/service/am/am.cpp:1516]: this probably should not be an error log

**Comment by BagpipesRbetter** [comment]
when will this be added to the full release

**Pull request merged**